src/components/BShuttle/BShuttle.vue

```diff
@@ -0,0 +1,480 @@
+    */
+    value: {
+      type: Array,
+      required: false,
+      default: function () { return [] }
+    }
+  },
+
+  data () {
+    return {
+      availableOptions: (this.options && this.options.length > 0) ? this.setOptions(this.options, this.value, 'available') : new Array,
+      selectedOptions: {'available': new Array, 'chosen': new Array},
+      chosenOptions: (this.value && this.options) ? this.setOptions(this.options, this.value, 'chosen') : new Array,
+      lastClick: 0,
+      id: null,
+      allToChosen: require('../../../static/doubleArrow.svg'),
+      allToAvailable: require('../../../static/doubleArrow-left.svg')
+    }
+  },
+
+  computed: {
+  },
+
+  created() {
+    this.$validator.extend('notEmpty', {
+      getMessage: field => 'The ' + field + ' input is required.',
+      validate: value => value.length > 0
+    });
+    this.$validator.attach({
+      name: this.validationId,
+      rules: 'notEmpty',
+      alias: this.chosenLabel
+    });
+
+  },
+
+  mounted() {
+    this.id = this._uid
+  },
+
+  methods: {
+
+    /**
+     * @param {Array<Object>} options
+     * @param {Array<Number>} selected
+     * @param {String} type
+     * this method sets up the initial state of the options
+     */
+    setOptions: function (options, selected, type) {
+      const availableOptions = new Array
+      const chosenOptions = new Array
+      options.forEach(function (option) {
+        if (selected === null) {
+          availableOptions.push(option)
+        } else if (selected.indexOf(option.id) === -1) {
+          availableOptions.push(option)
+        } else {
+          chosenOptions.push(option)
+        }
+      })
+      if (type === 'available') {
+        return this.sortById(availableOptions)
+      } else if (type === 'chosen') {
+        return this.sortById(chosenOptions)   
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * @param {String} optionType
+     * this determines if an option has been highlighted and
+     * sets the correct class.
+     */
+    optionsClass: function (option, optionType) {
+      return {
+        'options': true,
+        'selected': this.selectedOptions[optionType].indexOf(option.id) > -1
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * @param {String} optionType
+     * @param {String} otherOptionType
+     * @param {Object} event
+     * this is fired when a line is clicked and updates
+     * selectedOptions accordingly.
+     */
+    onOptionClick (option, optionType, otherOptionType, event) {
+      if (this.selectedOptions[otherOptionType].length > 0) { this.selectedOptions[otherOptionType] = new Array }
+      if (event.shiftKey && (this.lastClick > 0)) {
+        this.multipleSelect(option, optionType)
+      } else if (event.altKey) {
+        this.selectedOptions[otherOptionType] = new Array
+        this.selectedOptions[optionType] = [option.id]
+        this.onMoveSelectedClick(optionType)
```

Comment:
So this is calling an event function that this is not attached to. This seems like there should be a move function that takes a set of options. The current move function is dependent on an outside variable and it causes this section to immediately select it then move it.

Reply:
This didn't actually change. Not sure why it's outdated